src/components/AutocompleteSearchBar/AutocompleteSearchBar.js

```diff
@@ -0,0 +1,161 @@
+/*
+ * This file is part of React-SearchKit.
+ * Copyright (C) 2018 CERN.
+ *
+ * React-SearchKit is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import _ from 'lodash'
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Input } from 'semantic-ui-react';
+
+class AutocompleteSearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.updateQueryString = this.props.updateQueryString;
+    this.updateSuggetions = this.props.updateSuggetions;
+    this.clearSuggestions = this.props.clearSuggestions;
+    this.state = {
+      currentValue: this.props.queryString || '',
+    };
+    this.renderElement = props.renderElement || this._renderElement;
+    this.renderSuggestions = props.renderSuggestions || this._renderSuggestions;
+    this.handleAutocompleteChange = props.handleAutocompleteChange || this._handleAutocompleteChange;
+  };
+
+  _renderSuggestions(querySuggestions) {
+    // Define actions
+    const onSuggestionSelected = async (suggestion) => {
+      await this.setState({
+        currentValue: suggestion,
+      });
+      this.clearSuggestions();
+      this.executeSearch();
+    };
+
+    //Render Element
+    if (querySuggestions.length === 0) {
+      return null;
+    }
+
+    return (
+      <ul>
+        { querySuggestions.map( text => (
+          <li 
+            onClick={ () => onSuggestionSelected(text) }
+            key={ text }
+          >{ text }</li>)
+        )}
+      </ul>
+    );
+  
+  };
+
+  _renderElement(
+    placeholder,
+    debounce,
+    debounce_time,
+    queryString,
+    querySuggestions,
+    onInputChange,
+    executeSearch
+  ) {
+    // Initialize values              
+    placeholder = placeholder || 'Type something';
+    debounce = debounce || false;
+    // Define actions
+    const onBtnSearchClick = (event, input) => {
+      executeSearch();
+    };
+    const onKeyPress = (event, input) => {
+      if (event.key === 'Enter') {
+        executeSearch();
+      }
+    };
+
+    //Render Element
+    return (
+      <div className='AutoCompleteText'>
+        <Input
+          action={{
+            content: 'Search',
+            onClick: onBtnSearchClick,
+          }}
+          fluid
+          placeholder={placeholder}
+          onChange={(event, { value }) => {
+            onInputChange(value, debounce, debounce_time);
+          }}
+          value={queryString}          
+          onKeyPress={onKeyPress}
+        />
+        { this.renderSuggestions(querySuggestions) }
+      </div>
+    );
+  }
+
+  _handleAutocompleteChange = (queryString, debounce, debounce_time) => {
+    if(queryString.length >= 3) {
+      if (debounce){
+        const debouncedSuggestions = _.debounce(
```

Comment:
May I suggest to define the function once to avoid to redefine the same each time?

There is no super nice solution, given that 1 param come from props. I see 2 options:
* a partial function
* define it in the constructor

Something like (a bit ugly?):
```
_suggest = value => (
    (debounceTime) => (
        _debounce(this...., debounceTime, { ... })
   )
)

_handleAutocompleteChange = (...) => {
    ...
    _suggest(value)(props.debounceTime);
}
```

Or constructor:
```
constructor(props) {
    ...
    this.suggest = _debounce(
        this.props.updateSuggetions,
        this.props.debounceTime,
        { leading: true, }
    )
}
```

Reply:
Moved all to the constructor. Since ``debounce`` and ``debounceTime`` are in "props" it can all be just taken in the constructor and the conditional assignment done there. Therefore, it also simplifies the handler function.